nixos/modules/installer/netboot/netboot.nix

```diff
@@ -18,17 +18,15 @@ with lib;
 
   };
 
-  config = {
-
-    boot.loader.grub.version = 2;
-
+  config = rec {
     # Don't build the GRUB menu builder script, since we don't need it
     # here and it causes a cyclic dependency.
     boot.loader.grub.enable = false;
 
     # !!! Hack - attributes expected by other modules.
-    system.boot.loader.kernelFile = "bzImage";
-    environment.systemPackages = [ pkgs.grub2 pkgs.grub2_efi pkgs.syslinux ];
```

Comment:
I'm going to restore these packages for not aarch64, but I don't think they build on aarch64. I wonder what that means for "# !!! Hack - attributes expected by other modules.".

Reply:
`git grep kernelTarget` already gives the answer to `"bzImage``, so whatever uses that should be refactored to use `kernelTarget`.

Reply:
```
nixos/modules/system/activation/top-level.nix
32:        "${config.system.boot.loader.kernelFile}";
173:    system.boot.loader.kernelFile = mkOption {

nixos/modules/installer/cd-dvd/iso-image.nix
269:    system.boot.loader.kernelFile = "bzImage";

nixos/modules/installer/cd-dvd/system-tarball-sheevaplug.nix
85:  system.boot.loader.kernelFile = "uImage";
153:    { source = config.boot.kernelPackages.kernel + "/" + config.system.boot.loader.kernelFile;
154:      target = "/boot/" + config.system.boot.loader.kernelFile;

nixos/modules/installer/cd-dvd/system-tarball-pc.nix
82:    [ { source = config.boot.kernelPackages.kernel + "/" + config.system.boot.loader.kernelFile;
83:        target = "/boot/" + config.system.boot.loader.kernelFile;

nixos/modules/installer/cd-dvd/system-tarball-fuloong2f.nix
139:    { source = config.boot.kernelPackages.kernel + "/" + config.system.boot.loader.kernelFile;
140:      target = "/boot/" + config.system.boot.loader.kernelFile;
155:  system.boot.loader.kernelFile = "vmlinux";

nixos/modules/installer/netboot/netboot.nix
30:    system.boot.loader.kernelFile = if pkgs.stdenv.system == "aarch64-linux"
90:      kernel ${system.boot.loader.kernelFile} init=${config.system.build.toplevel}/init ${toString config.boot.kernelParams}

nixos/modules/system/boot/loader/generations-dir/generations-dir.nix
62:    system.boot.loader.kernelFile = platform.kernelTarget;

nixos/modules/system/boot/loader/raspberrypi/raspberrypi.nix
46:    system.boot.loader.kernelFile = platform.kernelTarget;
```

I'm not sure I want to take this on in this PR 😅

Reply:
Eh ... might as well ...

Reply:
Well you could change it to `system.boot.loader.kernelFile = platform.kernelTarget;` and call it a day.

Reply:
As for the packages, I think `grub2_efi` should build on AArch64.

BTW, don't you need to use `mkIf` to not end up with infinite recursion?

Reply:
I don't, systemPackages doesn't impact pkgs here. Fixed up the kernelTarget issue, added grub2_efi on aarch64.